Contributions/Using_Linux_in_a_Safe_System.md

```diff
@@ -0,0 +1,348 @@
+  what is being monitored, either directly or indirectly.
+
+  If this is not acceptable, then additional requirements must be formulated,
+  and additional monitoring (or other means of mitigation) must be deployed.
+
+  With regard to the example, having a monitor that detects excessive skew in
+  the period used by the application to ping, will not ensure that other parts of
+  the application are free from interference and that they will function correctly.
+
+  Never the less, carefully choosing how/where to implement the monitor and how to
+  prime it can avoid expensive, tedious, error-prone involvement of large chunks
+  of the kernel architecture in supporting safety claims.
+
+  It is usually far more effective to implement multi-stage monitoring
+  mechanisms, to ensure that the core component remains simple and easy to qualify,
+  while additional complexity of the validation can be achieved by chaining multiple
+  monitoring processes, terminating at the bottom with a very simple one.
+
+* **Watchdogs / Safety Islands / Other Exception Contexts**
+
+  Continuing from the previous point, there are three typical design approaches,
+  for implementing the monitoring.
+  
+  *  **Watchdog**
+
+     A watchdog is a well known (simple) method for imposing mandatory behaviour on
+     a system; the basic idea is to implement a "dead-man button": if a component
+     is not producing a predefined stimuli, periodically, the watchdog will
+     inderogably trigger a predefined event.
+     Typically a hardware watchdog is a peripheral that enforces one of the
+     following policies:
+     *  NMI: the watchdog generates a Not Maskable Interrupt, which forces a core
+        to execute a predefined interrupt handler. The handler runs with its own
+        sanitsied context, and can be assumed to be in a sane/safe/secure state.
+        However, it retains also the access to the system state.
+
+        One could rely on the NMI interrupt service routine to take over and
+        initiate a reset, or power down, or whatever is required.
+     *  reset: it usually consists in the electrical state of the system being
+        taken to its default values, paired with software re-initialisation.
+     *  power down: power is cut off, at least from the components associated
+        to safety requirements.
+
+     This is fairly boilerplate information, but it is worth mentioning that
+     safety requirements apply not only to software but to hardware as well.
+     The Watchdog IP must be safety qualified, to be relied on.
+
+     However, for the watchdog to support safety claims, the following must
+     be proven:
+     *  the initialisation of the watchdog can be considered sufficiently safe
+        (same or better safety level than the required one)
+     *  it must be impossible for a single failure in the system to make it
+        ineffective; this case covers:
```

Comment:
impossible???

I'd rather say "the probably of a failure in the watchdog itself or of a failure propagating from the system to the watchdog shall be coherent with the safety integrity level of the safety claim supported by the watchdog"